node/web/api/orderHistoryApiController.js

```diff
@@ -0,0 +1,25 @@
+module.exports.name = 'orderHistoryApiController';
+module.exports.dependencies = ['router', 'orderHistoryRepo', 'exceptions'];
+module.exports.factory = function (router, orderHistoryRepo, exceptions) {
+    'use strict';
+    router.get('/api/orderHistory/:userId', function (req, res) {
+		var userId = req.params.userId;
+		orderHistoryRepo.get(req.params.userId, function (err, orders) {
+            if (err) {
+                exceptions.throwException(err);
+                res.status(400);
+                return;
+            }
+
+			for (var i = 0; i < orders.length; i++) {
+            	var sum = 0;
+            	for (var j = 0; j < orders[i].books.length; j++) {
+            		sum += orders[i].books[j].price;
+				}
+				orders[i].total = sum;
```

Comment:
Also - it's not going to hurt the value, but it's not necessary to set the total in each loop. Here's an example that uses a functional interface to get the same job done:

```javascript
orders.total = 0;

orders.map(function (order) {
    order.books.map(function (book) {
        orders.total += book.price;
    });
});
```